build/charts/antrea/templates/agent/daemonset.yaml

```diff
@@ -101,6 +101,10 @@ spec:
             # binaries that need to be skipped for installation, e.g. "portmap, bandwidth".
             - name: SKIP_CNI_BINARIES
               value: {{ join "," .Values.cni.skipBinaries | quote }}
+            {{- if .Values.agent.dontLoadKernelModules }}
+            - name: SKIP_LOADING_KERNEL_MODULES
+              value: "1"
+            {{- end }}
```

Comment:
should it stop mounting host-lib-modules in this case?

Reply:
that's a good point, we don't need it. The con is that it introduces one more difference between the 2 "modes", but that's not a big deal. I will make the change and run some sanity checks.